src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -513,6 +655,98 @@ public function addAllowedValues($option, $allowedValues)
     }
 
     /**
+     * Adds allowed values for one or more options.
+     *
+     * First argument may be a name or an array of option names.
+     *
+     * You can pass a constant of Options interface as first argument to target
+     * a group of options. Supported values are:
+     *
+     *   - Options::ALL Defined and extra options
+     *   - Options::DEFINED
+     *   - Options::NESTED
+     *   - Options::EXTRA
+     *
+     * If a nested option name is passed it will apply to all nested options.
+     * You can prevent it by passing Options::NONE as fourth argument.
+     *
+     * Instead of passing values, you may also pass a closures with the
+     * following signature:
+     *
+     *     function ($value) {
+     *         // return true or false
+     *     }
+     *
+     * The closure receives the value as argument and should return true to
+     * accept the value and false to reject the value.
+     *
+     * @param string|string[]|int $optionNames   One or more option names
+     *                                           or an Options constant
+     * @param mixed               $allowedValues One or more accepted values
+     *                                           or closures
+     * @param bool                $replace       Whether to replace previous
+     *                                           values
+     * @param int                 $nested        This method is recursive for
+     *                                           nested options which is the
+     *                                           default. Pass Options::NONE
+     *                                           to change it.
+     *
+     * @return OptionsResolver This instance
+     *
+     * @throws UndefinedOptionsException If an option is undefined
+     * @throws AccessException           If called from a lazy option or normalizer
+     */
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [The method addAllowedValuesForAll() has a Cyclomatic Complexity of 12. The configured cyclomatic complexity threshold is 10.](https://www.codacy.com/app/git_9/symfony/file/2332052707/issues/source?bid=3207821&fileBranchId=3207869#l699)